tests/Microsoft.Identity.Test.Core.UIAutomation/MobileTestHelper.cs

```diff
@@ -74,8 +77,14 @@ public void AcquireTokenSilentTestHelper(ITestController controller, LabResponse
             AcquireTokenInteractiveHelper(controller, labResponse, CoreUiTestConstants.UiBehaviorLogin);
             VerifyResult(controller);
 
-            //acquire token for 2nd resource with refresh token
-            SetInputData(controller, labResponse.AppId, CoreUiTestConstants.DefaultScope, CoreUiTestConstants.UiBehaviorLogin);
+            controller.Tap(CoreUiTestConstants.SelectUser, XamarinSelector.ByAutomationId);
+            controller.Tap(labResponse.User.Upn);
+
+            if (_platform == Xamarin.UITest.Platform.iOS)
+            {
+                controller.Tap("Done");
```

Comment:
This seems to magical...

Reply:
? on iOS the tabs are on the bottom, so if we have picker on the screen, we have to hit the done button in order to see the tabs again. 

---
In reply to: [296153282](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1226#discussion_r296153282) [](ancestors = 296153282)

Reply:
Sorry, I know what this does and you've explained it to me. It's magical because it behaves differently on iOS than on Android and because it has absolutely nothing to do with our test. Most of the logic in the tests are about pressing buttons, filling forms and switching switches. It's impossible to understand what a test does by reading its source code, and it's also very difficult to maintain these tests.